src/pocketmine/plugin/PluginManager.php

```diff
@@ -362,39 +370,80 @@ public function loadPlugins(string $directory, array $newLoaders = null){
+		$required = $this->server->getApiVersion();
 
 		foreach($versions as $version){
-			//Format: majorVersion.minorVersion.patch (3.0.0)
-			//    or: majorVersion.minorVersion.patch-devBuild (3.0.0-alpha1)
-			if($version !== $serverString){
-				$pluginApi = array_pad(explode("-", $version), 2, ""); //0 = version, 1 = suffix (optional)
+			if(self::compareApi($required, $version)){
+				return true;
+			}
+		}
 
-				if(strtoupper($pluginApi[1]) !== strtoupper($serverApi[1])){ //Different release phase (alpha vs. beta) or phase build (alpha.1 vs alpha.2)
-					continue;
-				}
+		return false;
+	}
 
-				$pluginNumbers = array_map("intval", array_pad(explode(".", $pluginApi[0]), 3, "0")); //plugins might specify API like "3.0" or "3"
+	/**
+	 * Returns whether a given API version is compatible with the required version.
+	 *
+	 * @param string $haveVersion
+	 * @param string $needVersion
+	 *
+	 * @return bool
+	 */
+	public static function compareApi(string $haveVersion, string $needVersion) : bool{
+		//Format: majorVersion.minorVersion.patch (3.0.0)
+		//    or: majorVersion.minorVersion.patch-devBuild (3.0.0-alpha1)
+		if($needVersion !== $haveVersion){
+			$have = array_pad(explode("-", $haveVersion), 2, "");
+			$need = array_pad(explode("-", $needVersion), 2, ""); //0 = version, 1 = suffix (optional)
+
+			if(strtoupper($need[1]) !== strtoupper($have[1])){ //Different release phase (alpha vs. beta) or phase build (alpha.1 vs alpha.2)
+				return false;
+			}
 
-				if($pluginNumbers[0] !== $serverNumbers[0]){ //Completely different API version
-					continue;
-				}
+			$haveNumbers = array_map("intval", explode(".", $have[0]));
+			$needNumbers = array_map("intval", array_pad(explode(".", $need[0]), 3, "0")); //plugins might specify API like "3.0" or "3"
 
-				if($pluginNumbers[1] > $serverNumbers[1]){ //If the plugin requires new API features, being backwards compatible
-					continue;
-				}
+			if($needNumbers[0] !== $haveNumbers[0]){ //Completely different API version
+				return false;
+			}
 
-				if($pluginNumbers[1] === $serverNumbers[1] and $pluginNumbers[2] > $serverNumbers[2]){ //If the plugin requires bug fixes in patches, being backwards compatible
-					continue;
-				}
+			if($needNumbers[1] > $haveNumbers[1]){ //If the plugin requires new API features, being backwards compatible
+				return false;
 			}
 
-			return true;
+			if($needNumbers[1] === $haveNumbers[1] and $needNumbers[2] > $haveNumbers[2]){ //If the plugin requires bug fixes in patches, being backwards compatible
+				return false;
+			}
 		}
 
-		return false;
+		return true;
+	}
+
+	/**
+	 * Searches the given list of API versions and filters out unnecessary ones.
+	 *
+	 * @param string ...$versions
+	 *
+	 * @return string[][] base version (string) => unnecessary extra versions (string[])
+	 */
+	public static function findRedundantApis(string ...$versions) : array{
+		$redundant = [];
+		usort($versions, 'version_compare');
```

Comment:
Not sure if `version_compare` really works.

Reply:
It's a bit more loose than I'm happy with, but our version format works with `version_compare()` just fine.

Reply:
I'd rather you add an actual compare function that wraps version_compare(). It's too easy to miss this in the future in case we change the versioning scheme somehow such that it isn't version_compare()-compatible.